test/test-helpers.js

```diff
@@ -168,6 +168,26 @@ export const useFakeEnvironment = function(assert) {
   });
   fakeEnvironment.clock = sinon.useFakeTimers();
   fakeEnvironment.xhr = sinon.useFakeXMLHttpRequest();
+
+  // Sinon 1.10.2 handles abort incorrectly (triggering the error event)
+  // Later versions fixed this but broke the ability to set the response
+  // to an arbitrary object (in our case, a typed array).
+  XMLHttpRequest.prototype = Object.create(XMLHttpRequest.prototype);
+  XMLHttpRequest.prototype.abort = function abort() {
+    this.aborted = true;
+    this.response = this.responseText = '';
+    this.errorFlag = true;
+    this.requestHeaders = {};
+    this.responseHeaders = {};
+
+    if (this.readyState > 0 && this.sendFlag) {
+      this.readyStateChange(4);
```

Comment:
With XMLHttpRequest aborts, it looks like the `readystatechange` event is not fired (source: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/abort ).